go.mod

```diff
@@ -1,6 +1,6 @@
 module knative.dev/func
 
-go 1.22.7
+go 1.23
```

Comment:
should we check w/ the rest of upstream, regarding this bump?

CC @dsimansk

Reply:
In theory, we would eventually move to toolchain `1.23.z` regardless. Personally I don't like `go.mod` minVersion bumps unless warranted by other dependencies or specific go feature from newer version.

Reply:
The latest buildpack requires go1.23. Also if I just wanted to update source-to-image we are at the same place: the latest source-to-image requires docker v0.27.0 and only latest buildpacks work with that.